cornedbeef/src/main/java/com/swiftkey/cornedbeef/PunchHoleCoachMark.java

```diff
@@ -88,13 +115,70 @@ protected void updateView(CoachMarkDimens<Integer> popupDimens,  CoachMarkDimens
                 horizontalPadding, verticalPadding + lowerGap);
     }
 
+    /**
+     * Move the punch hole from left to right of the target view, unless the
+     * width of the target view is smaller than the diameter of the punch hole
+     * in which case the circle will be centered and the animation is pointless.
+     */
+    @TargetApi(Build.VERSION_CODES.HONEYCOMB)
+    private void animateHorizontalTranslation() {
+        if (canHaveHorizontalTranslation() && mHorizontalAnimator == null) {
+            final int leftMostPosition = mTargetViewLoc[0] + (int) mRelCircleRadius;
+            final int rightMostPosition = mTargetViewLoc[0] + mTargetView.getWidth() - (int) mRelCircleRadius;
+            final int animationStartX = isRtlConfig() ? rightMostPosition : leftMostPosition;
+            final int animationEndX = isRtlConfig() ? leftMostPosition : rightMostPosition;
+            mHorizontalAnimator = ObjectAnimator.ofInt(
+                    mPunchHoleView,
+                    "circleCenterX",
+                    animationStartX,
+                    animationEndX);
+            mHorizontalAnimator.addUpdateListener(
+                    new ValueAnimator.AnimatorUpdateListener() {
+                        @Override
+                        public void onAnimationUpdate(ValueAnimator animation) {
+                            mPunchHoleView.invalidate();
+                        }
+                    });
+            // the animation lasts as long as the coachmark is on screen
```

Comment:
I think this is very specific to this animation (I did it to test quicker), and we could have separate values for the animation duration and the coachmark timeout. We can set them both to the same value for our particular case.